### PR TITLE
Two new principles (actually one existing one recycled) for devices.

### PR DESCRIPTION
**Pull request by cynthia**
Written in a breakout today, this attempts to address #39 and #308.

***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/320.html" title="Last updated on Sep 15, 2021, 8:13 AM UTC (abd8832)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/320/0e3164c...abd8832.html" title="Last updated on Sep 15, 2021, 8:13 AM UTC (abd8832)">Diff</a>

### CODE REVIEW

**Review by LeaVerou — APPROVED** [review]

File: index.bs [hunk 1]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
...
+Try to design new native capabilities being brought to the web based on functionality.
 
-When adapting native operating system APIs for the web,
-make sure the new web APIs are designed with web platform principles in mind.
+You will probably not be able to directly translate
+an API available to native applications
+to a web API.
+
+Instead, consider the functionality available from the native API,
+and the user needs it addresses,
```
Comment (kenchris) [inline index.bs:1974]: addressed

File: index.bs [hunk 2]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
...
+and design an API that meets those user needs,
+even if the implementation depends on the existing native API.
+
+Be particularly careful about exposing
+the exact lifecycle and data structures of the underlying native APIs.
```
Comment (kenchris) [inline index.bs:1979]: I wonder if we want to say why. That probably makes it easier to understand. Like native APIs differ, and new ones might appear in the future that works quite differently. It is quite common to just deprecated APIs in native land, but not on the web

File: index.bs [hunk 3]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
...
 they may need to for web compatibility reasons.
 
-<h3 id="wrapper-apis">Native APIs don't typically translate well to the web</h3>
+<h3 id="usecase-oriented-apis">Design based on functionality, not the underlying API or hardware</h3>
 
+Try to design new native capabilities being brought to the web based on functionality.
 
-When adapting native operating system APIs for the web,
-make sure the new web APIs are designed with web platform principles in mind.
+You will probably not be able to directly translate
```
Comment (kenchris) [inline index.bs:1969]: Avoid directly translating an existing native API to the web.

File: index.bs [hunk 4]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
 expose a full list of devices in an [=implementation-defined=] order,
 they may need to for web compatibility reasons.
 
-<h3 id="wrapper-apis">Native APIs don't typically translate well to the web</h3>
+<h3 id="usecase-oriented-apis">Design based on functionality, not the underlying API or hardware</h3>
 
+Try to design new native capabilities being brought to the web based on functionality.
```
Comment (kenchris) [inline index.bs:1967]: I am not a fan of the "try to" Just say "Design"

Comment (kenchris) [reply index.bs:1967]: based on -> around their functionality

File: index.bs [hunk 5]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
...
-When adapting native operating system APIs for the web,
-make sure the new web APIs are designed with web platform principles in mind.
+You will probably not be able to directly translate
+an API available to native applications
+to a web API.
+
+Instead, consider the functionality available from the native API,
+and the user needs it addresses,
+and design an API that meets those user needs,
+even if the implementation depends on the existing native API.
```
Comment (kenchris) [inline index.bs:1976]: if? Maybe "as"

File: index.bs [hunk 6]
```diff
@@ -1962,24 +1962,55 @@ Note: While APIs should not
...
+
+Be particularly careful about exposing
+the exact lifecycle and data structures of the underlying native APIs.
+When possible, consider flexibility for new hardware.
+
+This means newly proposed APIs should be designed
+with careful consideration on how they are intended to be used
+rather than the underlying hardware, device, or native API
```
Comment (kenchris) [inline index.bs:1984]: rather than *how* the .... works today

**Comment by cynthia** [comment]
Just for reference, most of this text was already in there.

**Review by hober — APPROVED** [review]
LGTM

File: index.bs (hunk 4 above)
Comment (torgo) [reply index.bs:1967]:
```suggestion
Design new native capabilities being brought to the web based on functionality.
```

File: index.bs (hunk 2 above)
Comment (torgo) [reply index.bs:1977]: We should address this in a followup PR - will discuss in https://github.com/w3ctag/design-principles/issues/333.

File: index.bs (hunk 5 above)
Comment (torgo) [reply index.bs:1974]: we decided no.

Comment (cynthia) [reply index.bs:1974]: Group consensus that "if" is correct.

**Review by torgo — APPROVED** [review]
lgtm

**Pull request merged**